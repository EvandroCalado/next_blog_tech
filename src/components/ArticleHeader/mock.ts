export default {
  id: '3',
  title: 'Desenvolvimento de aplicativos web escaláveis com Next.js',
  excerpt:
    'Se você está procurando um framework poderoso para desenvolver aplicativos web escaláveis, o Next.js pode ser a escolha certa para você. O Next.js é um framework baseado em React que oferece recursos avançados, como renderização no lado do servidor e roteamento dinâmico, para ajudá-lo a criar aplicativos web modernos e de alto desempenho. Neste post, exploraremos as principais características do Next.js e veremos como usá-lo para construir aplicativos web escaláveis.',
  createdAt: '2023-03-27T17:22:39.749Z',
  cover:
    'https://res.cloudinary.com/dnhsdenji/image/upload/v1679929576/twitter_card_f0df8b6a20.png',
  author: {
    displayName: 'Evandro Calado',
    slug: 'evandro-calado',
  },
  categories: [
    {
      attributes: {
        displayName: 'React',
        slug: 'react',
      },
    },
    {
      attributes: {
        displayName: 'Next',
        slug: 'next',
      },
    },
  ],
};
