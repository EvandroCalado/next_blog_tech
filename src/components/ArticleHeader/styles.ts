import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};

      @media ${theme.media.small} {
        font-size: ${theme.font.sizes.medium};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};

    @media ${theme.media.small} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
