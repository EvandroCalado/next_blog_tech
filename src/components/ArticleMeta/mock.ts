export default {
  createdAt: '2023-03-27T17:22:39.749Z',
  author: {
    displayName: 'Evandro Calado',
    slug: 'evandro-calado',
  },
  categories: [
    {
      attributes: {
        displayName: 'React',
        slug: 'react',
      },
    },
    {
      attributes: {
        displayName: 'Next',
        slug: 'next',
      },
    },
  ],
};
