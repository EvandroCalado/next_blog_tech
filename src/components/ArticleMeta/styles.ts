import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child)::after {
      content: ', ';
    }

    span:first-child {
      margin-right: 0.5rem;
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
