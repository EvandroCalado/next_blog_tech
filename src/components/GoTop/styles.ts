import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    background-color: ${theme.colors.primary};
    color: ${theme.colors.white};
    position: fixed;
    bottom: 3rem;
    right: 5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    z-index: 6;
    text-decoration: none;
    transition: all 300ms ease-in-out;

    &:hover {
      color: ${theme.colors.secondary};
    }
  `}
`;
