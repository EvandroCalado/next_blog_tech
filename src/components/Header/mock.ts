import { HeaderProps } from '.';

export default {
  blogName: 'Evandro Calado',
  blogDescription:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Unde harum optio, eius quia quas molestias ipsum modi sed ullam exercitationem.',
  logo: 'https://res.cloudinary.com/dnhsdenji/image/upload/v1679924887/110628201_ae0753864b.jpg',
  showText: true,
} as HeaderProps;
