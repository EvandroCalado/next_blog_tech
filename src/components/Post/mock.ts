import { data } from '../../api/dados.json';

const { title, excerpt, content, cover, author, categories, createdAt } =
  data.posts.data[0].attributes;

const { id } = data.posts.data[0];

export default {
  id,
  title,
  excerpt,
  createdAt,
  content,
  cover: cover.data.attributes.url,
  author: author.data.attributes,
  categories: categories.data,
};
