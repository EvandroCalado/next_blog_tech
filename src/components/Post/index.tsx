import ArticleHeader, { ArticleHeaderProps } from '../ArticleHeader';
import HtmlContent from '../HtmlContent';
import { PostContainer } from '../PostContainer';
import * as Styled from './styles';

export type PostProps = ArticleHeaderProps & {
  content: string;
};
const Post = ({
  title,
  author,
  slug,
  categories,
  content,
  cover,
  createdAt,
  excerpt,
  id,
}: PostProps) => {
  return (
    <Styled.Wrapper>
      <PostContainer size="full">
        <ArticleHeader
          author={author}
          categories={categories}
          title={title}
          excerpt={excerpt}
          id={id}
          cover={cover}
          createdAt={createdAt}
          slug={slug}
        />
      </PostContainer>
      <PostContainer size="content">
        <HtmlContent html={content} />
      </PostContainer>
    </Styled.Wrapper>
  );
};

export default Post;
