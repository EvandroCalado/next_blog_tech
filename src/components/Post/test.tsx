import Post from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

import mock from './mock';
import { formatDate } from '../utils/format-date';

describe('<Post />', () => {
  it('should render header, excerpt, cover, metadata and post', () => {
    const { container } = renderTheme(<Post {...mock} />);
    expect(
      screen.getByRole('heading', { name: mock.title }),
    ).toBeInTheDocument();
    expect(screen.getByRole('img', { name: mock.slug })).toBeInTheDocument();
    expect(screen.getByText(formatDate(mock.createdAt))).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
