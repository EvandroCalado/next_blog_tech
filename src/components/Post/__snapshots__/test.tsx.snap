// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<Post /> should render header, excerpt, cover, metadata and post 1`
] = `
.c3 {
  color: #303030;
  line-height: 1.2;
  font-size: 6.4rem;
  font-weight: 900;
  text-transform: none;
  text-align: center;
}

.c1 {
  padding-bottom: 4rem;
  margin-bottom: 4rem;
  border-bottom: 0.1rem solid #ddd;
}

.c1 .c2 {
  margin: 0;
  margin-bottom: 2.4rem;
}

.c4 {
  margin: 2.4rem 0;
  font-size: 2.4rem;
}

.c5 {
  width: 100%;
  display: block;
  margin-bottom: 2.4rem;
}

.c6 {
  font-size: 1.6rem;
  color: #aaa;
  font-style: italic;
}

.c6 .categories span:not(:last-child)::after {
  content: ', ';
}

.c6 span:first-child {
  margin-right: 0.5rem;
}

.c6 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c6 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c9 {
  font-size: 2.4rem;
}

.c9 p {
  margin: 4.8rem 0;
}

.c9 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c9 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c9 img {
  max-width: 100%;
}

.c9 .image {
  background-color: #ddd;
  line-height: 0;
  margin: 4.8rem 0;
}

.c9 .image figcaption {
  font-size: 1.6rem;
  padding: 1.6rem;
  text-align: center;
  line-height: 1.3rem;
}

.c9 .image-style-side {
  float: right;
  margin: 2.4rem;
}

.c9 hr {
  border: none;
  border-bottom: 0.1rem solid #ddd;
}

.c9 ul,
.c9 ol {
  margin: 4.8rem;
}

.c9 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c9 table {
  width: 100%;
  border-collapse: collapse;
}

.c9 table td,
.c9 table th {
  padding: 1.6rem;
  border: 0.1rem solid #ddd;
}

.c8 pre[class*='language-'],
.c8 code[class*='language-'] {
  color: #d4d4d4;
  font-size: 13px;
  text-shadow: none;
  font-family: Menlo,Monaco,Consolas,'Andale Mono','Ubuntu Mono','Courier New',monospace;
  direction: ltr;
  text-align: left;
  white-space: pre;
  word-spacing: normal;
  word-break: normal;
  line-height: 1.5;
  -moz-tab-size: 4;
  -o-tab-size: 4;
  tab-size: 4;
  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  hyphens: none;
}

.c8 pre[class*='language-']::selection,
.c8 code[class*='language-']::selection,
.c8 pre[class*='language-'] *::selection,
.c8 code[class*='language-'] *::selection {
  text-shadow: none;
  background: #264f78;
}

.c8 pre[class*='language-'] {
  padding: 1em;
  margin: 0.5em 0;
  overflow: auto;
  background: #1e1e1e;
}

.c8:not(pre) > code[class*='language-'] {
  padding: 0.1em 0.3em;
  border-radius: 0.3em;
  color: #db4c69;
  background: #1e1e1e;
}

.c8 .namespace {
  opacity: 0.7;
}

.c8 .token.doctype .token.doctype-tag {
  color: #569cd6;
}

.c8 .token.doctype .token.name {
  color: #9cdcfe;
}

.c8 .token.comment,
.c8 .token.prolog {
  color: #6a9955;
}

.c8 .token.punctuation,
.c8 .language-html .language-css .token.punctuation,
.c8 .language-html .language-javascript .token.punctuation {
  color: #d4d4d4;
}

.c8 .token.property,
.c8 .token.tag,
.c8 .token.boolean,
.c8 .token.number,
.c8 .token.constant,
.c8 .token.symbol,
.c8 .token.inserted,
.c8 .token.unit {
  color: #b5cea8;
}

.c8 .token.selector,
.c8 .token.attr-name,
.c8 .token.string,
.c8 .token.char,
.c8 .token.builtin,
.c8 .token.deleted {
  color: #ce9178;
}

.c8 .language-css .token.string.url {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c8 .token.operator,
.c8 .token.entity {
  color: #d4d4d4;
}

.c8 .token.operator.arrow {
  color: #569cd6;
}

.c8 .token.atrule {
  color: #ce9178;
}

.c8 .token.atrule .token.rule {
  color: #c586c0;
}

.c8 .token.atrule .token.url {
  color: #9cdcfe;
}

.c8 .token.atrule .token.url .token.function {
  color: #dcdcaa;
}

.c8 .token.atrule .token.url .token.punctuation {
  color: #d4d4d4;
}

.c8 .token.keyword {
  color: #569cd6;
}

.c8 .token.keyword.module,
.c8 .token.keyword.control-flow {
  color: #c586c0;
}

.c8 .token.function,
.c8 .token.function .token.maybe-class-name {
  color: #dcdcaa;
}

.c8 .token.regex {
  color: #d16969;
}

.c8 .token.important {
  color: #569cd6;
}

.c8 .token.italic {
  font-style: italic;
}

.c8 .token.constant {
  color: #9cdcfe;
}

.c8 .token.class-name,
.c8 .token.maybe-class-name {
  color: #4ec9b0;
}

.c8 .token.console {
  color: #9cdcfe;
}

.c8 .token.parameter {
  color: #9cdcfe;
}

.c8 .token.interpolation {
  color: #9cdcfe;
}

.c8 .token.punctuation.interpolation-punctuation {
  color: #569cd6;
}

.c8 .token.boolean {
  color: #569cd6;
}

.c8 .token.property,
.c8 .token.variable,
.c8 .token.imports .token.maybe-class-name,
.c8 .token.exports .token.maybe-class-name {
  color: #9cdcfe;
}

.c8 .token.selector {
  color: #d7ba7d;
}

.c8 .token.escape {
  color: #d7ba7d;
}

.c8 .token.tag {
  color: #569cd6;
}

.c8 .token.tag .token.punctuation {
  color: #808080;
}

.c8 .token.cdata {
  color: #808080;
}

.c8 .token.attr-name {
  color: #9cdcfe;
}

.c8 .token.attr-value,
.c8 .token.attr-value .token.punctuation {
  color: #ce9178;
}

.c8 .token.attr-value .token.punctuation.attr-equals {
  color: #d4d4d4;
}

.c8 .token.entity {
  color: #569cd6;
}

.c8 .token.namespace {
  color: #4ec9b0;
}

.c8 pre[class*='language-javascript'],
.c8 code[class*='language-javascript'],
.c8 pre[class*='language-jsx'],
.c8 code[class*='language-jsx'],
.c8 pre[class*='language-typescript'],
.c8 code[class*='language-typescript'],
.c8 pre[class*='language-tsx'],
.c8 code[class*='language-tsx'] {
  color: #9cdcfe;
}

.c8 pre[class*='language-css'],
.c8 code[class*='language-css'] {
  color: #ce9178;
}

.c8 pre[class*='language-html'],
.c8 code[class*='language-html'] {
  color: #d4d4d4;
}

.c8 .language-regex .token.anchor {
  color: #dcdcaa;
}

.c8 .language-html .token.punctuation {
  color: #808080;
}

.c8 pre[class*='language-'] > code[class*='language-'] {
  position: relative;
  z-index: 1;
}

.c8 .line-highlight.line-highlight {
  background: #f7ebc6;
  box-shadow: inset 5px 0 0 #f7d87c;
  z-index: 0;
}

.c0 {
  width: 100%;
  margin: 0 auto;
  padding: 0 3.2rem;
  max-width: 120rem;
}

.c7 {
  width: 100%;
  margin: 0 auto;
  padding: 0 3.2rem;
  max-width: 80rem;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4rem;
  }
}

@media (max-width:360px) {
  .c1 .c2 {
    font-size: 2.4rem;
  }
}

@media (max-width:360px) {
  .c4 {
    font-size: 1.6rem;
  }
}

@media (max-width:768px) {
  .c9 {
    font-size: 2rem;
  }
}

@media print {
  .c8 pre[class*='language-'],
  .c8 code[class*='language-'] {
    text-shadow: none;
  }
}

@media (max-width:768px) {
  .c0 {
    padding: 1.6rem;
  }
}

@media (max-width:768px) {
  .c7 {
    padding: 1.6rem;
  }
}

<div>
  <article
    class=""
  >
    <div
      class="c0"
    >
      <header
        class="c1"
      >
        <h1
          class="c2 c3"
        >
          Desenvolvimento de aplicativos web escaláveis com Next.js
        </h1>
        <p
          class="c4"
        >
          Se você está procurando um framework poderoso para desenvolver aplicativos web escaláveis, o Next.js pode ser a escolha certa para você. O Next.js é um framework baseado em React que oferece recursos avançados, como renderização no lado do servidor e roteamento dinâmico, para ajudá-lo a criar aplicativos web modernos e de alto desempenho. Neste post, exploraremos as principais características do Next.js e veremos como usá-lo para construir aplicativos web escaláveis.
        </p>
        <img
          alt="desenvolvimento-de-aplicativos-web-escaláveis-com-next.js"
          class="c5"
          src="https://res.cloudinary.com/dnhsdenji/image/upload/v1679929576/twitter_card_f0df8b6a20.png"
        />
        <div
          class="c6"
        >
          <p>
            <span>
              Por
            </span>
            <a
              href="/author/evandro-calado"
            >
              Evandro Calado
            </a>
            <span
              class="separator"
            >
               | 
            </span>
            <time
              datetime="2023-03-27T17:22:39.749Z"
            >
              27 de mar. de 2023
            </time>
            <span
              class="separator"
            >
               | 
            </span>
            <span
              class="categories"
            >
              <span>
                <a
                  href="/category/react"
                >
                  React
                </a>
              </span>
              <span>
                <a
                  href="/category/next"
                >
                  Next
                </a>
              </span>
            </span>
          </p>
        </div>
      </header>
    </div>
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        <div
          class="c9"
        >
          <h2>
            O que é o Next.js?
          </h2>
          <p>
            O Next.js é um framework de desenvolvimento web baseado em React que foi criado pela equipe da Vercel. Ele oferece muitos recursos e funcionalidades para tornar a criação de aplicativos web mais fácil e rápida. Alguns dos recursos mais notáveis   do Next.js incluem:
          </p>
          <ul>
            <li>
              Renderização no lado do servidor (SSR) e na renderização do lado do cliente (CSR)
            </li>
            <li>
              Roteamento dinâmico
            </li>
            <li>
              Pré-renderização estática e dinâmica
            </li>
            <li>
              Suporte a CSS Modules
            </li>
            <li>
              Suporte a API externas
            </li>
          </ul>
          <p>
            Esses recursos tornam o Next.js uma escolha popular para a criação de aplicativos web escaláveis e de alto desempenho.
          </p>
          <hr />
          <figure
            class="image"
          >
            <a
              href="https://nextjs.org/"
            >
              <img
                alt="next example"
                sizes="100vw"
                src="https://res.cloudinary.com/dnhsdenji/image/upload/v1679937482/next_ed0a265ac1.jpg"
                srcset="https://res.cloudinary.com/dnhsdenji/image/upload/v1679937482/thumbnail_next_ed0a265ac1.jpg 245w,https://res.cloudinary.com/dnhsdenji/image/upload/v1679937482/small_next_ed0a265ac1.jpg 500w,https://res.cloudinary.com/dnhsdenji/image/upload/v1679937482/medium_next_ed0a265ac1.jpg 750w,https://res.cloudinary.com/dnhsdenji/image/upload/v1679937482/large_next_ed0a265ac1.jpg 1000w,"
              />
            </a>
            <figcaption>
              Next Js Example
            </figcaption>
          </figure>
          <h2>
            Como começar com o Next.js
          </h2>
          <p>
            Para começar a usar o 
            <a
              href="https://nextjs.org/"
            >
              Next.js
            </a>
            , você precisará instalá-lo. A maneira mais fácil de fazer isso é usando o 
            <a
              href="https://nodejs.org/en"
            >
              Node.js
            </a>
             e o gerenciador de pacotes NPM. Execute o seguinte comando no terminal para instalar o Next.js:
          </p>
          <pre
            class="language-javascript"
            tabindex="0"
          >
            <code
              class="language-javascript"
            >
              npm install next react react
              <span
                class="token operator"
              >
                -
              </span>
              dom

            </code>
          </pre>
          <p>
            Depois de instalado, você pode criar um novo aplicativo Next.js usando o comando create-next-app. Execute o seguinte comando no terminal para criar um novo aplicativo Next.js:
          </p>
          <pre
            class="language-javascript"
            tabindex="0"
          >
            <code
              class="language-javascript"
            >
              npx create
              <span
                class="token operator"
              >
                -
              </span>
              next
              <span
                class="token operator"
              >
                -
              </span>
              app my
              <span
                class="token operator"
              >
                -
              </span>
              app

            </code>
          </pre>
          <p>
            Isso criará um novo aplicativo Next.js em uma pasta chamada my-app. Você pode entrar na pasta e iniciar o servidor de desenvolvimento usando os seguintes comandos:
          </p>
          <pre
            class="language-javascript"
            tabindex="0"
          >
            <code
              class="language-javascript"
            >
              cd my
              <span
                class="token operator"
              >
                -
              </span>
              app
npm run dev

            </code>
          </pre>
          <h2>
            Páginas estáticas e dinâmicas
          </h2>
          <p>
            O Next.js oferece suporte a dois tipos de páginas: páginas estáticas e páginas dinâmicas.
          </p>
          <hr />
          <figure
            class="image image-style-side"
          >
            <img
              alt="next example"
              sizes="100vw"
              src="https://res.cloudinary.com/dnhsdenji/image/upload/v1679937482/next_ed0a265ac1.jpg"
              srcset="https://res.cloudinary.com/dnhsdenji/image/upload/v1679937482/thumbnail_next_ed0a265ac1.jpg 245w,https://res.cloudinary.com/dnhsdenji/image/upload/v1679937482/small_next_ed0a265ac1.jpg 500w,https://res.cloudinary.com/dnhsdenji/image/upload/v1679937482/medium_next_ed0a265ac1.jpg 750w,https://res.cloudinary.com/dnhsdenji/image/upload/v1679937482/large_next_ed0a265ac1.jpg 1000w,"
            />
            <figcaption>
              Next component
            </figcaption>
          </figure>
          <h3>
            Páginas estáticas
          </h3>
          <p
            style="margin-left:0px;"
          >
            As páginas estáticas são páginas que não mudam, independentemente de quem as solicita. Elas são pré-renderizadas no momento da construção e servidas como arquivos HTML estáticos para cada solicitação. Isso pode ajudar a melhorar o desempenho do aplicativo e reduzir o tempo de carregamento.
          </p>
          <p>
            Para criar uma página estática no Next.js, crie um arquivo 
            <strong>
              .js
            </strong>
             ou 
            <strong>
              .tsx
            </strong>
             na pasta 
            <strong>
              pages
            </strong>
            . Por exemplo, para criar uma página estática chamada 
            <strong>
              about
            </strong>
            , crie um arquivo chamado 
            <strong>
              about.js
            </strong>
             ou 
            <strong>
              about.tsx
            </strong>
             na pasta pages. Em seguida, adicione o seguinte código:
          </p>
          <p>
             
          </p>
          <pre
            class="language-javascript"
            tabindex="0"
          >
            <code
              class="language-javascript"
            >
              <span
                class="token keyword"
              >
                function
              </span>
               
              <span
                class="token function"
              >
                About
              </span>
              <span
                class="token punctuation"
              >
                (
              </span>
              <span
                class="token punctuation"
              >
                )
              </span>
               
              <span
                class="token punctuation"
              >
                {
              </span>
              
  
              <span
                class="token keyword"
              >
                return
              </span>
               
              <span
                class="token operator"
              >
                &lt;
              </span>
              h1
              <span
                class="token operator"
              >
                &gt;
              </span>
              About
              <span
                class="token operator"
              >
                &lt;
              </span>
              <span
                class="token operator"
              >
                /
              </span>
              h1
              <span
                class="token operator"
              >
                &gt;
              </span>
              

              <span
                class="token punctuation"
              >
                }
              </span>
              


              <span
                class="token keyword"
              >
                export
              </span>
               
              <span
                class="token keyword"
              >
                default
              </span>
               About

            </code>
          </pre>
          <p>
            Isso criará uma página estática com o título "About".
          </p>
          <h3>
            Páginas dinâmicas
          </h3>
          <p>
            As páginas dinâmicas são páginas que são renderizadas no servidor com base em dados específicos da solicitação. Por exemplo, uma página de perfil de usuário pode ser uma página dinâmica que é renderizada com base no nome de usuário da solicitação.
          </p>
          <p>
            <br />
             
          </p>
        </div>
        ;
      </div>
    </div>
  </article>
</div>
`;
